gnome-initial-setup/pages/endless-eula/gis-endless-eula-viewer.c

```diff
@@ -142,14 +152,14 @@ get_terms_document (void)
    */
   language = setlocale (LC_MESSAGES, NULL);
   locale_variants = g_get_locale_variants (language);
-  path = find_terms_document_for_languages ((const gchar * const *)locale_variants);
-  g_strfreev (locale_variants);
+  driver = GIS_DRIVER (g_application_get_default ());
+  product_name = gis_driver_get_product_name (driver);
 
-  if (path == NULL)
-    {
-      /* Now use "C" as a fallback. */
-      path = find_terms_document_for_language ("C");
-    }
+  path = find_terms_document_for_languages (product_name, (const gchar * const *)locale_variants);
+  if (path == NULL && g_strcmp0 ("eos", product_name) != 0)
```

Comment:
You could use `!gis_driver_is_product()` here, but do you really need that extra check? Without it, it just means that if we had an `eos` product but no ToS for `eos`, we'd check twice to make that determination. However that doesn't really happen since there always is an `eos` ToS...

Reply:
Ok, if assuming we always trust that there's going to be an `eos` ToS, I agree.